Sources/AppCoinsSDK/Domain/Entities/Purchase.swift

```diff
@@ -192,10 +203,15 @@ public class Purchase: Codable {
                             switch result {
                             case .success(let purchases):
                                 purchaseList += purchases
-                            case .failure(let failure):
-                                if failure == .failed { error = .systemError }
-                                else if failure == .noInternet { error = .networkError }
-                                else { error = .unknown }
+                            case .failure(let productServiceError):
+                                switch productServiceError {
+                                case .failed(let message, let description, let request):
+                                    error = AppCoinsSDKError.systemError(debugInfo: DebugInfo(message: message, description: description, request: request))
+                                case .noInternet(let message, let description, let request):
+                                    error = AppCoinsSDKError.networkError(debugInfo: DebugInfo(message: message, description: description, request: request))
+                                case .purchaseVerificationFailed(let message, let description, let request):
+                                    error = AppCoinsSDKError.unknown(debugInfo: DebugInfo(message: message, description: description, request: request))
```

Comment:
In this service, the error will never be an AAA error. I handled it this way because the switch case needs to be exhaustive. Would you prefer that I use default instead?